### PR TITLE
Adds Move Psychic Noise

### LINKED ISSUE #3844: Psychic Noise missing functionality
### Description

title

### Version

upcoming (Edge)

### Upcoming Version

_No response_

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Closes #3844

### CODE REVIEW

**Comment by cfmnephrite** [comment]
Given that Psychic Noisecs effect is a secondary effect blocked by Sheer Force and Shield Dust, do you think you could encapsulate it as a move effect? If you don't feel like doing that, then you need to check for those two abilities in the script.

File: data/battle_scripts_1.s [hunk 1]
```diff
@@ -370,6 +370,17 @@ gBattleScriptsForMoveEffects::
 	.4byte BattleScript_EffectHit                     @ EFFECT_FICKLE_BEAM
 	.4byte BattleScript_EffectHit                     @ EFFECT_BLIZZARD
 	.4byte BattleScript_EffectHit                     @ EFFECT_RAIN_ALWAYS_HIT
+	.4byte BattleScript_EffectPsychicNoise            @ EFFECT_PSYCHIC_NOISE
+
+BattleScript_EffectPsychicNoise::
+	call BattleScript_EffectHit_Ret
+	tryfaintmon BS_TARGET
+	jumpifability BS_TARGET_SIDE, ABILITY_AROMA_VEIL, BattleScript_AromaVeilProtects
+	sethealblock BattleScript_ButItFailed
```
[the remark this answers is not in the document]
Comment (AlexOn1ine) [reply data/battle_scripts_1.s:379]: will fix

**Comment by AlexOn1ine** [comment]
> Given that Psychic Noisecs effect is a secondary effect blocked by Sheer Force and Shield Dust, do you think you could encapsulate it as a move effect? If you don't feel like doing that, then you need to check for those two abilities in the script.

will do

**Comment by AlexOn1ine** [comment]
reviews applied

File: data/battle_scripts_1.s [hunk 2]
```diff
@@ -370,6 +370,12 @@ gBattleScriptsForMoveEffects::
 	.4byte BattleScript_EffectHit                     @ EFFECT_FICKLE_BEAM
 	.4byte BattleScript_EffectHit                     @ EFFECT_BLIZZARD
 	.4byte BattleScript_EffectHit                     @ EFFECT_RAIN_ALWAYS_HIT
+	.4byte BattleScript_EffectHit                     @ EFFECT_PSYCHIC_NOISE
+
+BattleScript_EffectPsychicNoise::
```
Comment (Bassoonian) [inline data/battle_scripts_1.s:375]: Why is this still defined (also in battle_scripts.h) if it's unused?

Comment (AlexOn1ine) [reply data/battle_scripts_1.s:375]: good question.

Comment (AlexOn1ine) [reply data/battle_scripts_1.s:374]: removed

**Review by Bassoonian — APPROVED** [review]

**Review by cfmnephrite — APPROVED** [review]

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -3793,6 +3793,21 @@ void SetMoveEffect(bool32 primary, bool32 certain)
                 }
                 SetMoveEffect(primary, certain);
                 break;
+            case MOVE_EFFECT_PSYCHIC_NOISE:
+                if (GetBattlerAbility(gEffectBattler) == ABILITY_AROMA_VEIL || GetBattlerAbility(BATTLE_PARTNER(gEffectBattler)) == ABILITY_AROMA_VEIL)
+                {
+                    gBattlerAbility = gEffectBattler;
+                    BattleScriptPush(gBattlescriptCurrInstr + 1);
+                    gBattlescriptCurrInstr = BattleScript_AromaVeilProtectsRet;
```
Comment (Skeli789) [inline src/battle_script_commands.c:3801]: Two things:
1 You should check ingame if there should be an Ability pop-up here. Usually when a secondary effect is blocked by an Ability (eg. Water Veil blocking burn) no Ability pop-up is shown -- the effect just doesn't happen.
2. If there _is_ supposed to be an Ability pop-up, it'll say the wrong Pokemon's Ability in the case where it's the partner's Ability that blocks it (you're setting `gBattlerAbility` to `gEffectBattler` regardless of the Ability holder)/.

Comment (AlexOn1ine) [reply src/battle_script_commands.c:3801]: 1. There is the function `IsAbilityOnSide` that is used for `ABILITY_AROMA_VEIL`. Would that suffice? I haven't figured out how to display the correct battler ability yet though. 
2. On showdown the ability pop up does show up. 
3. Correct. Should have been obvious 🫣

**Pull request merged**